doc/persistence.rst

```diff
@@ -61,10 +52,36 @@ Compressed joblib pickles
 Setting the `compress` argument to `True` in :func:`joblib.dump` will allow to
 save space on disk:
 
-  >>> joblib.dump(to_persist, filename, compress=True)  # doctest: +ELLIPSIS
-  ['...test.pkl']
+  >>> joblib.dump(to_persist, filename + '.z', compress=True)  # doctest: +ELLIPSIS
+  ['...test.pkl.z']
+
+By default, `joblib.dump` uses the zlib compression method as it gives the best
+tradeoff between speed and disk space. It also supports other compression
+formats such as gzip, bz2, lzma and xz:
+
+  >>> # Dumping in a gzip compressed file using a compress level of 3.
+  >>> joblib.dump(to_persist, filename + '.gz', compress=('gzip', 3))  # doctest: +ELLIPSIS
+  ['...test.pkl.gz']
+  >>> joblib.load(filename + '.gz')
+  [('a', [1, 2, 3]), ('b', array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))]
+
+When loading a compressed pickle, joblib will automatically select the right
```

Comment:
Maybe this is a bit too advanced/too much of an edge-case to be worth mentionning in the doc.